File: scripts/morse.rb
#!/usr/bin/ruby

#
# Use: ./morse.rb
# Heitor Gouvêa - hi@heitorgouvea.me

require 'morse'

def main
  if ARGV[0]
    $decodeMorse = Morse.decode(ARGV[0])

    puts $decodeMorse
  end
end

main()
exit

File: crypto/morse.pl
#!/usr/bin/perl

#
# Use: ./morse.pl "-- --- .-. ... ."
# Heitor Gouvêa - hi@heitorgouvea.me

use 5.010;
use strict;
use warnings;
use Text::Morse;

sub main {
  my $string = $ARGV[0];

  if ($string) {
    my $morse = new Text::Morse;

    print scalar($morse -> Decode($string));
  }
}

main();
exit;
